<!DOCTYPE html>
<!-- Generated by pkgdown: do not edit by hand --><html lang="en">
<head>
<meta http-equiv="Content-Type" content="text/html; charset=UTF-8">
<meta charset="utf-8">
<meta http-equiv="X-UA-Compatible" content="IE=edge">
<meta name="viewport" content="width=device-width, initial-scale=1, shrink-to-fit=no">
<title>Visualizing biological sequences with gt • gtseq</title>
<script src="deps/jquery-3.6.0/jquery-3.6.0.min.js"></script><meta name="viewport" content="width=device-width, initial-scale=1, shrink-to-fit=no">
<link href="deps/bootstrap-5.3.1/bootstrap.min.css" rel="stylesheet">
<script src="deps/bootstrap-5.3.1/bootstrap.bundle.min.js"></script><link href="deps/font-awesome-6.5.2/css/all.min.css" rel="stylesheet">
<link href="deps/font-awesome-6.5.2/css/v4-shims.min.css" rel="stylesheet">
<script src="deps/headroom-0.11.0/headroom.min.js"></script><script src="deps/headroom-0.11.0/jQuery.headroom.min.js"></script><script src="deps/bootstrap-toc-1.0.1/bootstrap-toc.min.js"></script><script src="deps/clipboard.js-2.0.11/clipboard.min.js"></script><script src="deps/search-1.0.0/autocomplete.jquery.min.js"></script><script src="deps/search-1.0.0/fuse.min.js"></script><script src="deps/search-1.0.0/mark.min.js"></script><!-- pkgdown --><script src="pkgdown.js"></script><meta property="og:title" content="Visualizing biological sequences with gt">
<meta name="description" content="Traditionally, analysis of related biological sequences through multiple sequence alignment (MSA) is done through standalone software programs like [Jalview](https://www.jalview.org/), [Rasmol](https://www.openrasmol.org/), and many others, or online web apps like [NCBIs msa viewer](https://www.ncbi.nlm.nih.gov/projects/msaviewer/). Recent additions like [`ggmsa`](https://yulab-smu.top/ggmsa/), Bioconductors [`msa`](https://www.bioconductor.org/packages/release/bioc/html/msa.html), among others, have brought visualization and analysis to the R ecosystem. But sequences are not often visualized in table packages like `gt`. This package facilitates bringing these traditional bioinformatic activities to R and `gt`.">
<meta property="og:description" content="Traditionally, analysis of related biological sequences through multiple sequence alignment (MSA) is done through standalone software programs like [Jalview](https://www.jalview.org/), [Rasmol](https://www.openrasmol.org/), and many others, or online web apps like [NCBIs msa viewer](https://www.ncbi.nlm.nih.gov/projects/msaviewer/). Recent additions like [`ggmsa`](https://yulab-smu.top/ggmsa/), Bioconductors [`msa`](https://www.bioconductor.org/packages/release/bioc/html/msa.html), among others, have brought visualization and analysis to the R ecosystem. But sequences are not often visualized in table packages like `gt`. This package facilitates bringing these traditional bioinformatic activities to R and `gt`.">
</head>
<body>
    <a href="#main" class="visually-hidden-focusable">Skip to contents</a>


    <nav class="navbar navbar-expand-lg fixed-top bg-light" data-bs-theme="light" aria-label="Site navigation"><div class="container">

    <a class="navbar-brand me-2" href="index.html">gtseq</a>

    <small class="nav-text text-muted me-auto" data-bs-toggle="tooltip" data-bs-placement="bottom" title="">0.0.1</small>


    <button class="navbar-toggler" type="button" data-bs-toggle="collapse" data-bs-target="#navbar" aria-controls="navbar" aria-expanded="false" aria-label="Toggle navigation">
      <span class="navbar-toggler-icon"></span>
    </button>

    <div id="navbar" class="collapse navbar-collapse ms-3">
      <ul class="navbar-nav me-auto">
<li class="nav-item"><a class="nav-link" href="reference/index.html">Reference</a></li>
<li class="nav-item dropdown">
  <button class="nav-link dropdown-toggle" type="button" id="dropdown-articles" data-bs-toggle="dropdown" aria-expanded="false" aria-haspopup="true">Articles</button>
  <ul class="dropdown-menu" aria-labelledby="dropdown-articles">
<li><a class="dropdown-item" href="articles/getting-started.html">Biological sequences in tables</a></li>
  </ul>
</li>
<li class="nav-item"><a class="nav-link" href="news/index.html">Changelog</a></li>
      </ul>
<ul class="navbar-nav">
<li class="nav-item"><form class="form-inline" role="search">
 <input class="form-control" type="search" name="search-input" id="search-input" autocomplete="off" aria-label="Search site" placeholder="Search for" data-search-index="search.json">
</form></li>
<li class="nav-item"><a class="external-link nav-link" href="https://github.com/wvictor14/gtseq/" aria-label="GitHub"><span class="fa fab fa-github fa-lg"></span></a></li>
      </ul>
</div>


  </div>
</nav><div class="container template-home">
<div class="row">
  <main id="main" class="col-md-9"><div class="section level1">
<div class="page-header"><h1 id="gtseq">gtseq<a class="anchor" aria-label="anchor" href="#gtseq"></a>
</h1></div>
<!-- badges: start -->

<p>The goal of <code>gtseq</code> is to provide smalle modular functions that help visualize biological sequences with <code>gt</code>.</p>
<p>This R package was developed as part of the 2025 posit table contest. See my <a href="https://github.com/rich-iannone/table-contest/discussions/10" class="external-link">submission</a> for details.</p>
<p>Traditionally, analysis of related biological sequences through multiple sequence alignment (MSA) is done through standalone software programs like <a href="https://www.jalview.org/" class="external-link">Jalview</a>, <a href="https://www.openrasmol.org/" class="external-link">Rasmol</a>, and many others, or online web apps like <a href="https://www.ncbi.nlm.nih.gov/projects/msaviewer/" class="external-link">NCBI’s msa viewer</a>. Recent additions like <a href="https://yulab-smu.top/ggmsa/" class="external-link"><code>ggmsa</code></a>, Bioconductor’s <a href="https://www.bioconductor.org/packages/release/bioc/html/msa.html" class="external-link"><code>msa</code></a>, among others, have brought visualization and analysis to the R ecosystem. But sequences are not often visualized in table packages like <code>gt</code>. This package facilitates bringing these traditional bioinformatic activities to R and <code>gt</code>.</p>
<div class="section level2">
<h2 id="installation">Installation<a class="anchor" aria-label="anchor" href="#installation"></a>
</h2>
<p>You can install the development version of gtseq from <a href="https://github.com/" class="external-link">GitHub</a> with:</p>
<div class="sourceCode" id="cb1"><pre class="downlit sourceCode r">
<code class="sourceCode R"><span><span class="co"># install.packages("pak")</span></span>
<span><span class="fu">pak</span><span class="fu">::</span><span class="fu"><a href="https://pak.r-lib.org/reference/pak.html" class="external-link">pak</a></span><span class="op">(</span><span class="st">"wvictor14/gtseq"</span><span class="op">)</span></span></code></pre></div>
</div>
<div class="section level2">
<h2 id="example">Example<a class="anchor" aria-label="anchor" href="#example"></a>
</h2>
<p>gtseq provides several functions to help create <code>gt</code> tables of biological sequences. Although the philosophy is to provide small modular functions to allow for high customizability, <code><a href="reference/gtseq.html">gtseq::gtseq</a></code> is a wrapper function provided to quickly and efficiently explore sequences.</p>
<p>Use <code>gtseq</code> on a <code>data.frame</code> with a column containing amino acid sequences:</p>
<div class="sourceCode" id="cb2"><pre class="downlit sourceCode r">
<code class="sourceCode R"><span><span class="kw"><a href="https://rdrr.io/r/base/library.html" class="external-link">library</a></span><span class="op">(</span><span class="va"><a href="http://victoryuan.com/gtseq/">gtseq</a></span><span class="op">)</span></span>
<span></span>
<span><span class="va">msa</span> <span class="op">|&gt;</span> <span class="fu"><a href="reference/gtseq.html">gtseq</a></span><span class="op">(</span>seq_column <span class="op">=</span> <span class="va">seq</span><span class="op">)</span></span></code></pre></div>
<p>For further customization use the lower-level functions:</p>
</div>
<div class="section level2">
<h2 id="split_sequences">
<code>split_sequences</code><a class="anchor" aria-label="anchor" href="#split_sequences"></a>
</h2>
<p><code>split_sequences</code> for structuring the data for <code>gt</code>, the output of this function retains the 1 sequence per row structure, but splits each element of the sequence data, for maximum flexibility for customization in <code>gt</code></p>
<div class="sourceCode" id="cb3"><pre class="downlit sourceCode r">
<code class="sourceCode R"><span><span class="kw"><a href="https://rdrr.io/r/base/library.html" class="external-link">library</a></span><span class="op">(</span><span class="va"><a href="https://gt.rstudio.com" class="external-link">gt</a></span><span class="op">)</span></span>
<span><span class="va">msa</span> <span class="op">|&gt;</span> <span class="fu"><a href="reference/split_sequences.html">split_sequences</a></span><span class="op">(</span>seq_column <span class="op">=</span> <span class="va">seq</span><span class="op">)</span></span></code></pre></div>
</div>
<div class="section level2">
<h2 id="apply_color_to_aa">
<code>apply_color_to_aa</code><a class="anchor" aria-label="anchor" href="#apply_color_to_aa"></a>
</h2>
<p><code>apply_color_to_aa</code> is for coloring amino acids by one of the included palettes</p>
<p>A function is returned, which takes an input and colors according to a given palette</p>
<div class="sourceCode" id="cb4"><pre class="downlit sourceCode r">
<code class="sourceCode R"><span><span class="fu"><a href="reference/apply_color_to_aa.html">apply_color_to_aa</a></span><span class="op">(</span><span class="st">"Chemistry"</span><span class="op">)</span><span class="op">(</span><span class="fu"><a href="https://rdrr.io/r/base/c.html" class="external-link">c</a></span><span class="op">(</span><span class="st">"A"</span>, <span class="st">"K"</span>, <span class="st">"L"</span>, <span class="st">"-"</span><span class="op">)</span><span class="op">)</span></span></code></pre></div>
<p>This function can be directly passed to <code><a href="https://gt.rstudio.com/reference/data_color.html" class="external-link">gt::data_color</a></code></p>
<div class="sourceCode" id="cb5"><pre class="downlit sourceCode r">
<code class="sourceCode R"><span><span class="va">msa</span> <span class="op">|&gt;</span></span>
<span>  <span class="fu"><a href="reference/split_sequences.html">split_sequences</a></span><span class="op">(</span>seq_column <span class="op">=</span> <span class="va">seq</span><span class="op">)</span> <span class="op">|&gt;</span></span>
<span>  <span class="fu"><a href="https://gt.rstudio.com/reference/gt.html" class="external-link">gt</a></span><span class="op">(</span><span class="op">)</span> <span class="op">|&gt;</span></span>
<span>  <span class="fu"><a href="https://gt.rstudio.com/reference/data_color.html" class="external-link">data_color</a></span><span class="op">(</span>columns <span class="op">=</span> <span class="fu"><a href="https://tidyselect.r-lib.org/reference/starts_with.html" class="external-link">contains</a></span><span class="op">(</span><span class="st">'pos_'</span><span class="op">)</span>, fn <span class="op">=</span> <span class="fu"><a href="reference/apply_color_to_aa.html">apply_color_to_aa</a></span><span class="op">(</span><span class="st">'Chemistry'</span><span class="op">)</span><span class="op">)</span></span></code></pre></div>
</div>
<div class="section level2">
<h2 id="get_consensus">
<code>get_consensus</code><a class="anchor" aria-label="anchor" href="#get_consensus"></a>
</h2>
<p><code>get_consensus</code> for adding consensus sequence and frequency data. Pass to either <code><a href="https://gt.rstudio.com/reference/grand_summary_rows.html" class="external-link">gt::grand_summary_rows</a></code> or <code><a href="https://gt.rstudio.com/reference/summary_rows.html" class="external-link">gt::summary_rows</a></code>.</p>
<div class="sourceCode" id="cb6"><pre class="downlit sourceCode r">
<code class="sourceCode R"><span><span class="va">msa</span> <span class="op">|&gt;</span></span>
<span>  <span class="fu"><a href="reference/split_sequences.html">split_sequences</a></span><span class="op">(</span>seq_column <span class="op">=</span> <span class="va">seq</span><span class="op">)</span> <span class="op">|&gt;</span></span>
<span>  <span class="fu"><a href="https://gt.rstudio.com/reference/gt.html" class="external-link">gt</a></span><span class="op">(</span><span class="op">)</span> <span class="op">|&gt;</span></span>
<span>  <span class="fu"><a href="https://gt.rstudio.com/reference/data_color.html" class="external-link">data_color</a></span><span class="op">(</span>columns <span class="op">=</span> <span class="fu"><a href="https://tidyselect.r-lib.org/reference/starts_with.html" class="external-link">contains</a></span><span class="op">(</span><span class="st">'pos_'</span><span class="op">)</span>, fn <span class="op">=</span> <span class="fu"><a href="reference/apply_color_to_aa.html">apply_color_to_aa</a></span><span class="op">(</span><span class="st">'Chemistry'</span><span class="op">)</span><span class="op">)</span> <span class="op">|&gt;</span></span>
<span>  <span class="fu"><a href="https://gt.rstudio.com/reference/grand_summary_rows.html" class="external-link">grand_summary_rows</a></span><span class="op">(</span></span>
<span>    columns <span class="op">=</span> <span class="fu"><a href="https://tidyselect.r-lib.org/reference/starts_with.html" class="external-link">contains</a></span><span class="op">(</span><span class="st">'pos_'</span><span class="op">)</span>,</span>
<span>    fns <span class="op">=</span> <span class="fu"><a href="https://rdrr.io/r/base/list.html" class="external-link">list</a></span><span class="op">(</span></span>
<span>      consensus <span class="op">=</span> <span class="op">~</span> <span class="fu"><a href="reference/get_consensus.html">get_consensus</a></span><span class="op">(</span><span class="va">.</span><span class="op">)</span></span>
<span>    <span class="op">)</span></span>
<span>  <span class="op">)</span></span></code></pre></div>
</div>
<div class="section level2">
<h2 id="generate_breaks">
<code>generate_breaks</code><a class="anchor" aria-label="anchor" href="#generate_breaks"></a>
</h2>
<p>Takes a vector and returns every nth element, based on <code><a href="https://scales.r-lib.org/reference/breaks_width.html" class="external-link">scales::breaks_width</a></code></p>
<div class="sourceCode" id="cb7"><pre class="downlit sourceCode r">
<code class="sourceCode R"><span><span class="kw"><a href="https://rdrr.io/r/base/library.html" class="external-link">library</a></span><span class="op">(</span><span class="va"><a href="https://dplyr.tidyverse.org" class="external-link">dplyr</a></span><span class="op">)</span></span>
<span><span class="va">cnames</span> <span class="op">&lt;-</span> <span class="va">msa</span> <span class="op">|&gt;</span></span>
<span>  <span class="fu"><a href="reference/split_sequences.html">split_sequences</a></span><span class="op">(</span>seq_column <span class="op">=</span> <span class="va">seq</span><span class="op">)</span> <span class="op">|&gt;</span></span>
<span>  <span class="fu"><a href="https://dplyr.tidyverse.org/reference/select.html" class="external-link">select</a></span><span class="op">(</span><span class="fu"><a href="https://tidyselect.r-lib.org/reference/starts_with.html" class="external-link">contains</a></span><span class="op">(</span><span class="st">'pos_'</span><span class="op">)</span><span class="op">)</span> <span class="op">|&gt;</span></span>
<span>  <span class="fu"><a href="https://rdrr.io/r/base/colnames.html" class="external-link">colnames</a></span><span class="op">(</span><span class="op">)</span></span>
<span><span class="va">breaks</span> <span class="op">&lt;-</span> <span class="fu"><a href="reference/generate_breaks.html">generate_breaks</a></span><span class="op">(</span><span class="va">cnames</span><span class="op">)</span></span>
<span><span class="va">breaks</span></span></code></pre></div>
<p>Which allows only applying a label to every nth position of the sequences with <code><a href="https://gt.rstudio.com/reference/cols_label_with.html" class="external-link">gt::cols_label_with</a></code></p>
<div class="sourceCode" id="cb8"><pre class="downlit sourceCode r">
<code class="sourceCode R"><span><span class="va">msa</span> <span class="op">|&gt;</span></span>
<span>  <span class="fu"><a href="reference/split_sequences.html">split_sequences</a></span><span class="op">(</span>seq_column <span class="op">=</span> <span class="va">seq</span><span class="op">)</span> <span class="op">|&gt;</span></span>
<span>  <span class="fu"><a href="https://gt.rstudio.com/reference/gt.html" class="external-link">gt</a></span><span class="op">(</span><span class="op">)</span> <span class="op">|&gt;</span></span>
<span>  <span class="fu"><a href="https://gt.rstudio.com/reference/data_color.html" class="external-link">data_color</a></span><span class="op">(</span>columns <span class="op">=</span> <span class="fu"><a href="https://tidyselect.r-lib.org/reference/starts_with.html" class="external-link">contains</a></span><span class="op">(</span><span class="st">'pos_'</span><span class="op">)</span>, fn <span class="op">=</span> <span class="fu"><a href="reference/apply_color_to_aa.html">apply_color_to_aa</a></span><span class="op">(</span><span class="st">'Chemistry'</span><span class="op">)</span><span class="op">)</span> <span class="op">|&gt;</span></span>
<span>  <span class="fu"><a href="https://gt.rstudio.com/reference/cols_label_with.html" class="external-link">cols_label_with</a></span><span class="op">(</span></span>
<span>    columns <span class="op">=</span> <span class="fu"><a href="https://tidyselect.r-lib.org/reference/starts_with.html" class="external-link">contains</a></span><span class="op">(</span><span class="st">'pos_'</span><span class="op">)</span>,</span>
<span>    fn <span class="op">=</span> <span class="op">~</span> <span class="fu"><a href="https://rdrr.io/r/base/ifelse.html" class="external-link">ifelse</a></span><span class="op">(</span><span class="va">.</span> <span class="op"><a href="https://rdrr.io/r/base/match.html" class="external-link">%in%</a></span> <span class="va">breaks</span>, <span class="fu"><a href="https://rdrr.io/r/base/grep.html" class="external-link">gsub</a></span><span class="op">(</span><span class="st">'pos_'</span>, <span class="st">''</span>, <span class="va">.</span><span class="op">)</span>, <span class="st">''</span><span class="op">)</span></span>
<span>  <span class="op">)</span></span></code></pre></div>
</div>
</div>
  </main><aside class="col-md-3"><div class="links">
<h2 data-toc-skip>Links</h2>
<ul class="list-unstyled">
<li><a href="https://github.com/wvictor14/gtseq/" class="external-link">Browse source code</a></li>
</ul>
</div>

<div class="license">
<h2 data-toc-skip>License</h2>
<ul class="list-unstyled">
<li><a href="LICENSE.html">Full license</a></li>
<li><small><a href="https://opensource.org/licenses/mit-license.php" class="external-link">MIT</a> + file <a href="LICENSE-text.html">LICENSE</a></small></li>
</ul>
</div>


<div class="citation">
<h2 data-toc-skip>Citation</h2>
<ul class="list-unstyled">
<li><a href="authors.html#citation">Citing gtseq</a></li>
</ul>
</div>

<div class="developers">
<h2 data-toc-skip>Developers</h2>
<ul class="list-unstyled">
<li>Victor Yuan <br><small class="roles"> Author, maintainer </small>   </li>
</ul>
</div>

<div class="dev-status">
<h2 data-toc-skip>Dev status</h2>
<ul class="list-unstyled">
<li><a href="https://github.com/wvictor14/gtseq/actions/workflows/R-CMD-check.yaml" class="external-link"><img src="https://github.com/wvictor14/gtseq/actions/workflows/R-CMD-check.yaml/badge.svg" alt="R-CMD-check"></a></li>
</ul>
</div>

  </aside>
</div>


    <footer><div class="pkgdown-footer-left">
  <p>Developed by Victor Yuan.</p>
</div>

<div class="pkgdown-footer-right">
  <p>Site built with <a href="https://pkgdown.r-lib.org/" class="external-link">pkgdown</a> 2.1.3.</p>
</div>

    </footer>
</div>





  </body>
</html>
